plugin-server/src/worker/ingestion/properties-updater.ts

```diff
@@ -115,19 +105,32 @@ export async function upsertGroup(
             }
 
             if (propertiesUpdate.updated) {
-                // :TRICKY: insertGroup will raise a RaceConditionError if group was inserted in-between fetch and this
-                const upsertMethod = group ? 'updateGroup' : 'insertGroup'
```

Comment:
while this is clever it makes it hard to use code editing/discovery tools effectively, e.g. find references

Reply:
let's then have those functions accept an object and do something like:

```js
const groupUpdateArgs = {
    teamId,
    groupTypeIndex,
    groupKey,
    createdAt,
    version,
    client,
    properties: propertiesUpdate.properties,
    propertiesLastUpdatedAt: propertiesUpdate.properties_last_updated_at,
    propertiesLastOperation: propertiesUpdate.properties_last_operation,
}

if (group) {
    await db.updateGroup(groupUpdateArgs)
} else {
    // :TRICKY: insertGroup will raise a RaceConditionError if group was inserted in-between fetch and this
    await db.insertGroup(groupUpdateArgs)
}
```

Reply:
this actually would increase total lines of code due to updateGroup & insertGroup needing to have the input defined differently + in the future we might want to change them independently for params, so thinking of not changing it.